src/sortStudents.ts

```diff
@@ -1,16 +1,78 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: Array<number>,
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
+}
+
+export enum SortOrder {
+  ascending = 'asc',
+  descending = 'desc',
+}
+
+type CompareFunc = (st1: Student, st2: Student) => number;
+
+function compareFuncGenerator(byProp: SortType
+  , order: SortOrder = SortOrder.ascending): CompareFunc {
+  let callback: CompareFunc;
+
+  switch (byProp) {
+    case SortType.Name:
+    case SortType.Surname:
+      callback = (st1, st2): number => {
+        return st1[byProp].localeCompare(st2[byProp]);
+      };
+      break;
+
+    case SortType.Age:
+    case SortType.Married:
+      callback = (st1, st2): number => {
+        return Number(st1[byProp]) - Number(st2[byProp]);
+      };
+      break;
+
+    case SortType.AverageGrade:
+      callback = (st1, st2): number => {
+        const st1GradesAvg = st1[byProp]
+          .reduce((prev, next) => prev + next) / st1[byProp].length;
```

Comment:
Додавай значення заумовчуванням до reduce
```suggestion
          .reduce((prev, next) => prev + next, 0) / st1[byProp].length;
```